Add more docs about notification plugins (#527)
Uninett/Argus: docs/notifications.rst

------------------------------------
Notifications and their destinations
------------------------------------

Notifications are sent with the help of notification plugins to destinations.

A notification plugin is a class that inherits from
``argus.notificationprofile.media.base.NotificationMedium``. It has a
``send(event, destinations, **kwargs)`` static method that does the actual
sending.

A destination is a user-specific and plugin-specific instance of the model
Destination. In the Destination there's a field ``settings`` that has the

> necessary configuration for the plugin.

Reviewer: It would be nice to add here in informal language what a destination is (e.g. a place to send messages to)

Notification plugins included on install
----------------------------------------

argus.notificationprofile.media.email.EmailNotification
.......................................................

This plugin is enabled by default.

This plugin uses the email server settings provided by Django itself to
configure the server.

The settings-field for an email-destination contains an
``email_address``-field. Hidden from the API, it also contains

> a ``synced``-field. which is used for some magic if the User model-instance has

Reviewer: ```suggestion
a ``synced``-field, which is used for some magic if the User model-instance has
```

> its ``email_address``-field set.

Reviewer: Since we mention in the sms part that the phone number is validated and one cannot test with an arbitrary string of numbers this should also be added here since `email_address` is also validated.

argus.notificationprofile.media.sms_as_email.SMSNotification
..............................................................

This plugin is **not** enabled by default.

This plugin is for systems where SMSes are sent via a magical email-address
(legacy-system support). For that reason it depends on the same Django email
server settings as the included EmailNotification-plugin, and in addition the
Argus-specific ``SMS_GATEWAY_ADDRESS``-setting, which is the magical

> email-address to send the SMS to.

Reviewer: ```suggestion
email-address to send the SMSes to.
```

> The settings-field for an SMS-destination contain only a ``phone_number``,

Reviewer: ```suggestion
The settings-field for an SMS-destination contains only a ``phone_number``,
```

The settings-field for an email-destination contains an
``email_address``-field. Hidden from the API, it also contains
a ``synced``-field, which is used for some magic if the User model-instance has
its ``email_address``-field set.

To validate the email address we use Django's own email validator.

argus.notificationprofile.media.sms_as_email.SMSNotification
............................................................

This plugin is **not** enabled by default.

This plugin is for systems where SMSes are sent via a magical email-address
(legacy-system support). For that reason it depends on the same Django email
server settings as the included EmailNotification-plugin, and in addition the
Argus-specific ``SMS_GATEWAY_ADDRESS``-setting, which is the magical
email-address to send the SMSes to.

> The phone number is suffixed to the localpart of the email-address.

Reviewer: ```suggestion
The phone number is suffixed to the local-part of the email-address.
```
